integrations/google_ai/tests/embedders/test_text_embedder.py

```diff
@@ -0,0 +1,379 @@
+
+
+# --- Warm Up Tests ---
+
+
+def test_warm_up_resolves_key_and_creates_client(embedder, mock_genai_client):
+    """Test warm_up resolves API key and initializes the client."""
+    mock_client_constructor, mock_client_instance = mock_genai_client
+
+    embedder.warm_up()
+
+    # Assert API key was resolved
+    assert embedder._api_key_resolved == "test-api-key"
+    # Assert Client constructor was called with the resolved key
+    # The Client constructor should be called with the actual resolved API key.
+    mock_client_constructor.assert_called_once_with(api_key="test-api-key")
+    # Assert the client instance is stored
+    assert embedder.client == mock_client_instance
+
+
+def test_warm_up_already_warmed_up(embedder, mock_genai_client):
+    """Test that warm_up doesn't re-initialize if called multiple times."""
+    mock_client_constructor, _ = mock_genai_client
+
+    embedder.warm_up()  # First call
+    embedder.warm_up()  # Second call
+
+    # Assert Client constructor was called only once
+    mock_client_constructor.assert_called_once()
+
+
+def test_warm_up_client_instantiation_fails(embedder, mock_genai_client):
+    """Test warm_up raises ValueError if client creation fails."""
+    mock_client_constructor, _ = mock_genai_client
+    # Configure the mock constructor to raise an exception
+    mock_client_constructor.side_effect = Exception("Client creation failed")
+
+    with pytest.raises(ValueError, match="Failed to configure Google AI client: Client creation failed"):
+        embedder.warm_up()
+
+
+# --- Run Tests ---
+
+
+def test_run_without_warm_up_raises(embedder):
+    """Test run raises RuntimeError if warm_up wasn't called."""
+    with pytest.raises(RuntimeError, match="The component has not been warmed up"):
+        embedder.run(texts=["test text"])
+
+
+def test_run_invalid_input_type_raises(embedder):
+    """Test run raises TypeError for invalid input."""
+    embedder.warm_up()  # Needs warm_up first
+    with pytest.raises(TypeError, match="GoogleAIGeminiTextEmbedder expects a List of strings"):
+        embedder.run(texts="not a list")  # type: ignore
+    with pytest.raises(TypeError, match="GoogleAIGeminiTextEmbedder expects a List of strings"):
+        embedder.run(texts=[1, 2, 3])  # type: ignore
+
+
+def test_run_empty_list(embedder):
+    """Test run with an empty list returns empty results."""
+    embedder.warm_up()
+    result = embedder.run(texts=[])
+    assert result == {"embedding": [], "meta": {"model": embedder.model, "task_type": embedder.task_type}}
+
+
+def test_run_api_call_success(embedder, mock_genai_client):
+    """Test a successful run call."""
+    mock_client_constructor, mock_client_instance = mock_genai_client
+    texts = ["text 1", "text 2"]
+    expected_embeddings = [[0.1, 0.2], [0.3, 0.4]]
+    # Configure the mock embed_content method to return a successful response
+    mock_client_instance.models.embed_content.return_value = {"embedding": expected_embeddings}
```

Comment:
This is the wrong mocking I was mentioning above. 

According to docs, a correct mock should be something like:

```python
mock_response = MagicMock()
mock_response.embeddings = None # or e.g. [[0.1, 0.2], [0.3, 0.4]]
mock_client_instance.models.embed_content.return_value = mock_response
```

Can you please update tests accordingly?